cli/cmd/resources.go

```diff
@@ -1,174 +1,195 @@
-					if !isDefault {
-						item.SetP("", "spec.default")
-					} else {
-						item.SetP("*", "spec.default")
-					}
-					return nil
-				},
-			}),
-			resourcemanager.WithDeleteEnabled("DataStore removed. Defaulting back to no-tracing mode"),
-		),
-	).
-	Register(
-		resourcemanager.NewClient(
-			httpClient,
-			"environment", "environments",
-			resourcemanager.WithTableConfig(resourcemanager.TableConfig{
-				Cells: []resourcemanager.TableCellConfig{
-					{Header: "ID", Path: "spec.id"},
-					{Header: "NAME", Path: "spec.name"},
-					{Header: "DESCRIPTION", Path: "spec.description"},
-				},
-			}),
-		),
-	).
-	Register(
-		resourcemanager.NewClient(
-			httpClient,
-			"transaction", "transactions",
-			resourcemanager.WithTableConfig(resourcemanager.TableConfig{
-				Cells: []resourcemanager.TableCellConfig{
-					{Header: "ID", Path: "spec.id"},
-					{Header: "NAME", Path: "spec.name"},
-					{Header: "VERSION", Path: "spec.version"},
-					{Header: "STEPS", Path: "spec.summary.steps"},
-					{Header: "RUNS", Path: "spec.summary.runs"},
-					{Header: "LAST RUN TIME", Path: "spec.summary.lastRun.time"},
-					{Header: "LAST RUN SUCCESSES", Path: "spec.summary.lastRun.passes"},
-					{Header: "LAST RUN FAILURES", Path: "spec.summary.lastRun.fails"},
-				},
-				ItemModifier: func(item *gabs.Container) error {
-					// set spec.summary.steps to the number of steps in the transaction
-					item.SetP(len(item.Path("spec.steps").Children()), "spec.summary.steps")
-
-					if err := formatItemDate(item, "spec.summary.lastRun.time"); err != nil {
-						return err
-					}
-
-					return nil
-				},
-			}),
-		),
-	).
-	Register(
-		resourcemanager.NewClient(
-			httpClient,
-			"test", "tests",
-			resourcemanager.WithTableConfig(resourcemanager.TableConfig{
-				Cells: []resourcemanager.TableCellConfig{
-					{Header: "ID", Path: "spec.id"},
-					{Header: "NAME", Path: "spec.name"},
-					{Header: "VERSION", Path: "spec.version"},
-					{Header: "TRIGGER TYPE", Path: "spec.trigger.type"},
-					{Header: "RUNS", Path: "spec.summary.runs"},
-					{Header: "LAST RUN TIME", Path: "spec.summary.lastRun.time"},
-					{Header: "LAST RUN SUCCESSES", Path: "spec.summary.lastRun.passes"},
-					{Header: "LAST RUN FAILURES", Path: "spec.summary.lastRun.fails"},
-					{Header: "URL", Path: "spec.url"},
-				},
-				ItemModifier: func(item *gabs.Container) error {
-					// set spec.summary.steps to the number of steps in the transaction
-					id, ok := item.Path("spec.id").Data().(string)
-					if !ok {
-						return fmt.Errorf("test id '%s' is not a string", id)
-					}
-
-					url := cliConfig.URL() + "/test/" + id
-					item.SetP(url, "spec.url")
-
-					if err := formatItemDate(item, "spec.summary.lastRun.time"); err != nil {
-						return err
-					}
-
-					return nil
-				},
-			}),
-		),
+var (
+	httpClient = &resourcemanager.HTTPClient{}
+
+	testPreprocessor = preprocessor.Test(cliLogger)
+	testClient       = resourcemanager.NewClient(
```

Comment:
this client is needed by the transaction preprocessor, so its here now to avoid a ciruclar initialization loop